commerce-service/src/main/java/com/liferay/commerce/internal/notification/CommerceDefinitionTermContributorRegistryImpl.java

```diff
@@ -0,0 +1,138 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.internal.notification;
+
+import com.liferay.commerce.order.CommerceDefinitionTermContributor;
+import com.liferay.commerce.order.CommerceDefinitionTermContributorRegistry;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerCustomizerFactory;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * @author Luca Pellizzon
+ */
+@Component(
+	immediate = true, service = CommerceDefinitionTermContributorRegistry.class
+)
+public class CommerceDefinitionTermContributorRegistryImpl
+	implements CommerceDefinitionTermContributorRegistry {
+
+	@Override
+	public List<CommerceDefinitionTermContributor>
+		getDefinitionTermContributorsByContributorKey(String key) {
+
+		List
+			<ServiceTrackerCustomizerFactory.ServiceWrapper
+				<CommerceDefinitionTermContributor>>
+					commerceDefinitionTermContributorWrappers =
+						_serviceTrackerMapByTermContributorKey.getService(key);
+
+		if (commerceDefinitionTermContributorWrappers == null) {
+			return Collections.emptyList();
+		}
+
+		List<CommerceDefinitionTermContributor>
+			commerceDefinitionTermContributors = new ArrayList<>();
+
+		for (ServiceTrackerCustomizerFactory.ServiceWrapper
+				<CommerceDefinitionTermContributor>
+					tableActionProviderServiceWrapper :
+						commerceDefinitionTermContributorWrappers) {
+
+			commerceDefinitionTermContributors.add(
+				tableActionProviderServiceWrapper.getService());
+		}
+
+		return commerceDefinitionTermContributors;
+	}
+
+	@Override
+	public List<CommerceDefinitionTermContributor>
+		getDefinitionTermContributorsByNotificationTypeKey(String key) {
+
+		List
+			<ServiceTrackerCustomizerFactory.ServiceWrapper
+				<CommerceDefinitionTermContributor>>
+					commerceDefinitionTermContributorWrappers =
+						_serviceTrackerMapByNotificationTypeKey.getService(key);
+
+		if (commerceDefinitionTermContributorWrappers == null) {
+			return Collections.emptyList();
+		}
+
+		List<CommerceDefinitionTermContributor>
+			commerceDefinitionTermContributors = new ArrayList<>();
+
+		for (ServiceTrackerCustomizerFactory.ServiceWrapper
+				<CommerceDefinitionTermContributor>
+					tableActionProviderServiceWrapper :
+						commerceDefinitionTermContributorWrappers) {
+
+			commerceDefinitionTermContributors.add(
+				tableActionProviderServiceWrapper.getService());
+		}
+
+		return commerceDefinitionTermContributors;
+	}
```

Comment:
It occurs to me that these methods are exactly the same except for the `ServiceTrackerMap` being used. It might be nice to consolidate the logic into a private method, and just pass in either the `ServiceTrackerMap` or the `commerceDefinitionTermContributorWrappers` to the method, so we only have to maintain it in one place.